### PR TITLE
Slim down on typed array allocations in conversion.ts

### PR DESCRIPTION
**Pull request by austinEng**
Improves deserialization from the case cache by at least 2x, with fewer GC events.

Issue: #

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/util/conversion.ts [hunk 1]
```diff
@@ -979,108 +983,104 @@ export interface ScalarBuilder {
...
-export function abstractFloat(value: number): Scalar {
-  const arr = new Float64Array([value]);
-  return new Scalar(TypeAbstractFloat, arr[0], arr);
+/** Create a Scalar of `type` by storing `value` as an element of `workingDataArray` and retrieving it. */
+function scalarFromValue(
+  type: ScalarType,
+  workingDataArray: TypedArrayBufferView,
+  value: number
+): Scalar {
+  workingDataU32[1] = 0;
```
Comment (ben-clayton) [inline src/webgpu/util/conversion.ts:992]: I assume this is to zero the array before assigning to (the aliased) `workingDataArray`? A comment here would be good.

File: src/webgpu/util/conversion.ts [hunk 2]
```diff
@@ -979,108 +983,104 @@ export interface ScalarBuilder {
...
+/** Create a Scalar of `type` by storing `value` as an element of `workingDataStoreArray` and
+ * reinterpreting it as an element of `workingDataLoadArray`.
+ */
+function scalarFromBits(
+  type: ScalarType,
+  workingDataStoreArray: TypedArrayBufferView,
+  workingDataLoadArray: TypedArrayBufferView,
+  bits: number
+): Scalar {
+  workingDataU32[1] = 0;
```
Comment (ben-clayton) [inline src/webgpu/util/conversion.ts:1007]: Same thing here.

**Review by zoddicus — APPROVED** [review]

File: src/webgpu/util/conversion.ts (hunk 2 above)
Comment (austinEng) [reply src/webgpu/util/conversion.ts:1012]: done

**Comment by ben-clayton** [comment]
LGTM

**Pull request merged**